Copy over the Instruction Formats text from John Hauser's docs (#247)
riscv/riscv-p-spec: P-ext-proposal.adoc

the double-wide instruction is equivalent to two single-wide instructions that
share the same scalar `rs2` (no `rs2+`), and must be performed simultaneously,
or in an order that does not overwrite `rs2` between the two operations.

<<<
== Instruction Formats

A few non-SIMD instructions of the P extension (including ABS, CLS, PACK,
REV, and REV16) are encoded alongside other ordinary RISC-V "scalar"
instructions in the major opcodes known as OP (binary code 0110011), OP-IMM
(0010011), and OP-IMM-32 (0011011).

All other Base P instructions are encoded either: (a) within major opcode OP-32
(binary code 0111011) with instruction bit 31 = 1; or (b) within OP-IMM-32 with

> bits 14:12 (func3) = binary 010, 100, or 110.

Reviewer: Suggestion to improve encoding readability 
```suggestion
instructions in the major opcodes known as OP (binary code `0110011`), OP-IMM
(`0010011`), and OP-IMM-32 (`0011011`).

All other Base P instructions are encoded either: (a) within major opcode OP-32
(binary code `0111011`) with instruction bit 31 = 1; or (b) within OP-IMM-32 with
bits 14:12 (func3) = binary `010`, `100`, or `110`.
```

> bits 14:12 (func3) = binary `010`, `100`, or `110`.

Author: ```suggestion
bits 14:12 (funct3) = binary `010`, `100`, or `110`.
```

]}
....

[wavedrom, ,svg]
....
{reg:[
    { bits: 7, name: 0x1b, attr: ['OP-IMM-32'] },
    { bits: 5, name: 'rd' },
    { bits: 3, name: 0x4 },
    { bits: 5, name: 'rs1' },
    { bits: 5, name: 'rs2' },
    { bits: 2, name: 'w' },
    { bits: 1, name: 0x1 },
    { bits: 3, name: 'f' },
    { bits: 1, name: 0x1 },
]}
....

[wavedrom, ,svg]
....
{reg:[
    { bits: 7, name: 0x3b, attr: ['OP-32'] },
    { bits: 5, name: 'rd' },
    { bits: 1, name: 0x0 },
    { bits: 1, name: 'x' },
    { bits: 1, name: 'x' },
    { bits: 5, name: 'rs1' },
    { bits: 5, name: 'rs2' },
    { bits: 2, name: 'w' },
    { bits: 4, name: 'f' },
    { bits: 1, name: 0x1 },
]}
....

[wavedrom, ,svg]
....
{reg:[
    { bits: 7, name: 0x3b, attr: ['OP-32'] },
    { bits: 5, name: 'rd' },
    { bits: 1, name: 0x1 },
    { bits: 1, name: 'x' },
    { bits: 1, name: 'x' },
    { bits: 5, name: 'rs1' },
    { bits: 5, name: 'rs2' },
    { bits: 2, name: 'w' },
    { bits: 1, name: 'R' },
    { bits: 3, name: 'f' },
    { bits: 1, name: 0x1 },
]}
....

Note that he first four formats are in major opcode OP-IMM-32, and the last two
are in OP-32. Of the OP-IMM-32 formats, those with func3 = binary `010` are

> mostly shifts left, while those with func3 = binary `100` are mostly shifts right.

Author: ```suggestion
are in OP-32. Of the OP-IMM-32 formats, those with funct3 = binary `010` are
mostly shifts left, while those with funct3 = binary `100` are mostly shifts right.
```
